routes/reviewsRouter.js

```diff
@@ -0,0 +1,123 @@
+const express = require('express');
+const router = express.Router();
+const Review = require('../models/schemas/reviews');
+
+router.get('/', (req, res) => {
+    Review.find({}, (err, reviews) => {
+        if (err) {
+            console.log(err);
+            res.status(500).json(err);
+        } else {
+            res.status(200).json(reviews);
+        }
+    });
+});
+
+router.post('/post', (req, res) => {
+    const { title, 
+        gameTitle, 
+        shortDescription, 
+        review, 
+        reviewerName, 
+        dateReviewed, 
+        reviewerScore }
+        = req.body;
+    new Review({ 
+        title,
+        gameTitle,
+        shortDescription,
+        review,
+        reviewerName,
+        dateReviewed,
+        reviewerScore
+    }).save((err, review) => {
+        if (err) {
+            console.log(err);
+            res.status(500).json({ status: 'Not OK', err });
+        } else {
+            res.status(200).json({ status: 'OK', review})
+        }
+    });
+});
+
+router.post('/search', (req, res) => {
+    const { title, 
+        gameTitle, 
+        shortDescription, 
+        review, 
+        reviewerName, 
+        dateReviewed, 
+        reviewerScore }
+        = req.body;
+    const query = {};
+    if (title) {
+        query.title = title;
+    }
+    if (gameTitle) {
+        query.gameTitle = gameTitle;
+    }
+    if (shortDescription) {
+        query.shortDescription = shortDescription;
+    }
+    if (review) {
+        query.review = review
+    }
+    if (reviewerName) {
+        query.reviewerName = reviewerName;
+    }
+    if (dateReviewed) {
+        query.dateReviewed = dateReviewed;
+    }
+    if (reviewerScore) {
+        query.reviewerScore = reviewerScore;
+    }
+    Review.find(query, (err, reviews) => {
+        if (err) {
+            console.log(err);
+            res.status(204).json([]);
```

Comment:
Remember to handle errors with the right status codes

Reply:
Do you think it's better to have a 500 code for the error and then 404 for `if (!reviews)`? I wrote it the same way in the usersRouter so I'll need to make changes to that also.

Reply:
Yeah, a 500 for if there is an error and a 404 if nothing comes back